resources/assets/js/mixins/services.js

```diff
@@ -55,6 +55,34 @@ export default {
         }
     },
     methods: {
+        serviceEndDate(s) {
+            if (!s.channels || !s.channels.length) {
+                return s.end_date;
+            }
+            return s.channels.map(channel => {
+                if (!channel.openinghours) {
+                    return null
+                }
+                return channel.openinghours.map(oh => oh.end_date)
+                    .sort((a, b) => {
+                        if (a < b) {
+                            return 1;
+                        }
+                        if (a > b) {
+                            return -1
+                        }
+                        return 0
+                    })[0]
+            }).sort((a, b) => {
+                if (a > b) {
+                    return 1;
+                }
+                if (a < b) {
+                    return -1
+                }
+                return 0
+            })[0]
```

Comment:
(Optional) Might be interesting to extract this snippet into a function to improve readability:

```js
(a, b) => {
  if (a > b) {
    return 1;
  }
  if (a < b) {
    return -1
  }
  return 0
}
```

Reply:
yes... it is a hideous beast isn't it....
I'll see if I can add them to the existing lib functions.